interfaces/mount/spec.go

```diff
@@ -206,3 +226,39 @@ func (spec *Specification) AddPermanentSlot(iface interfaces.Interface, slot *sn
 	}
 	return nil
 }
+
+// AddParallelInstanceMapping records mappings of snap instance directories
+//
+// When the snap is installed with an instance key, set up it's mount namespace
+// such that it appears as a non-instance key snap. This ensures compatibility
+// with code making assumptions about
+// $SNAP{,_DATA,_COMMON,_USER_DATA,_USER_COMMON} locations. That is, given a
+// snap foo_bar, the mappings added are:
+//
+// - /snap/foo_bar -> /snap/foo
+// - /var/snap/foo_bar -> /var/snap/foo
+// - $HOME/snap/foo_bar -> $HOME/snap/foo
+func (spec *Specification) AddParallelInstanceMapping(info *snap.Info) {
+	if info.InstanceKey == "" {
+		return
+	}
+
+	// /snap/foo_bar -> /snap/foo
+	spec.AddParallelInstanceMountEntry(osutil.MountEntry{
+		Name:    path.Join(dirs.CoreSnapMountDir, info.InstanceName()),
```

Comment:
Note-to-self the "core" here refers o the snap mount directory (`/snap`) as seen from inside mount namespaces or on core devices. This is in contrast to the snap mount directory on the host distribution which may be either `/snap` or `/var/lib/snapd/snap`